### PR TITLE
fix(2152): Support PR analysis for coverage info [3]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

When users open PRs, we should take advantage of Sonar PR analysis so new projects are not created.

## Objective

This PR adds logic to get scope, jobId, and jobName for PRs for Sonar coverage plugin.

## References

Related to #2152 

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/coverage/token.js [hunk 1]
```diff
@@ -18,32 +18,21 @@ module.exports = config => ({
                 security: [{ token: [] }]
             }
         },
-        handler: (request, reply) => {
+        handler: async (request, reply) => {
             const buildCredentials = request.auth.credentials;
-            const { jobFactory } = request.server.app;
             const { jobId } = buildCredentials;
```
Comment (jithine) [inline plugins/coverage/token.js:23]: `buildCredentials` contains `prParentJobId` & `isPR`, you don't need to fetch job again in `getCoverageConfig` for them

File: plugins/coverage/index.js [hunk 2]
```diff
@@ -13,6 +15,62 @@ const tokenRoute = require('./token');
...
+                    coverageConfig.annotations =
+                        job.permutations[0] && job.permutations[0].annotations ? job.permutations[0].annotations : {};
+                }
+
+                // If scope is job and job is pull request, set jobId and jobName
+                if (coverageConfig.annotations[COVERAGE_SCOPE_ANNOTATION] === 'job' && isPR) {
+                    const prNameMatch = job.name.match(prRegex);
+
+                    coverageConfig.jobId = job.prParentJobId;
+                    coverageConfig.jobName = prNameMatch && prNameMatch.length > 1 ? prNameMatch[2] : job.name;
```
Comment (jithine) [inline plugins/coverage/index.js:57]: is `coverageConfig.jobName` required ? we don't even need to fetch job here, all info is present in `buildCredentials` and can be passed here.

File: plugins/coverage/index.js [hunk 3]
```diff
@@ -13,6 +15,62 @@ const tokenRoute = require('./token');
...
+     * @return {Promise}                        Resolves to config object
+     */
+    server.expose('getCoverageConfig', ({ scope, jobId, prNum }) => {
+        const { jobFactory } = server.root.app;
+        const coverageConfig = {
+            jobId,
+            annotations: scope ? { [COVERAGE_SCOPE_ANNOTATION]: scope || null } : {},
+            prNum
+        };
+        const prRegex = /^PR-(\d+)(?::([\w-]+))?$/;
```
[the remark this answers is not in the document]
Comment (tkyi) [reply plugins/coverage/index.js]: no, the regex in data schema will group by `PR-xxx` vs `xxx`, which is what I want
https://github.com/screwdriver-cd/data-schema/blob/master/config/regex.js#L46

File: plugins/coverage/index.js (hunk 2 above)
Comment (tkyi) [reply plugins/coverage/index.js:57]: yes, we'll need it for `projectName` for `/info` endpoint

File: test/plugins/coverage.test.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ const mockery = require('mockery');
 
 sinon.assert.expose(assert, { prefix: '' });
 
-describe('coverage plugin test', () => {
+describe.only('coverage plugin test', () => {
```
Comment (jithine) [inline test/plugins/coverage.test.js:10]: remove only

**Pull request merged**